crates/uv/src/commands/project/sync.rs

```diff
@@ -76,6 +76,16 @@ pub(super) async fn do_sync(
     cache: &Cache,
     printer: Printer,
 ) -> Result<(), ProjectError> {
+    // Validate that the Python version is supported by the lockfile.
+    if let Some(requires_python) = lock.requires_python() {
+        if !requires_python.contains(venv.interpreter().python_version()) {
+            return Err(ProjectError::RequiresPython(
+                venv.interpreter().python_version().clone(),
+                requires_python.clone(),
+            ));
```

Comment:
Should this be a warning?

Reply:
Probably not, I feel like for a lockfile it makes sense to fail?